File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/person/update/PersonUpdate.js
define([ "stjs", "as/dto/common/update/FieldUpdateValue", "as/dto/common/update/IdListUpdateValue", "as/dto/webapp/update/WebAppSettingsUpdateValue" ], function(stjs, FieldUpdateValue,
		IdListUpdateValue, WebAppSettingsUpdateValue) {
	var PersonUpdate = function() {
		this.spaceId = new FieldUpdateValue();
		this.active = new FieldUpdateValue();
	};
	stjs.extend(PersonUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.person.update.PersonUpdate';
		constructor.serialVersionUID = 1;
		prototype.userId = null;
		prototype.spaceId = null;
		prototype.webAppSettings = null;
		prototype.active = null;

		prototype.getObjectId = function() {
			return this.getUserId();
		};
		prototype.getUserId = function() {
			return this.userId;
		};
		prototype.setUserId = function(userId) {
			this.userId = userId;
		};
		prototype.getSpaceId = function() {
			return this.spaceId;
		};
		prototype.setSpaceId = function(spaceId) {
			this.spaceId.setValue(spaceId);
		};
		prototype.getWebAppSettings = function(webAppId) {
			if (webAppId === undefined) {
				return this.webAppSettings;
			} else {
				if (this.webAppSettings == null) {
					this.webAppSettings = {};
				}

				var updateValue = this.webAppSettings[webAppId];

				if (updateValue == null) {
					updateValue = new WebAppSettingsUpdateValue();
					this.webAppSettings[webAppId] = updateValue;
				}

				return updateValue;
			}
		};
		prototype.isActive = function() {
			return this.active;
		};
		prototype.activate = function() {
			this.active.setValue(true);
		};
		prototype.deactivate = function() {
			this.active.setValue(false);
		};
	}, {
		userId : "IPersonId",
		spaceId : {
			name : "FieldUpdateValue",
			arguments : [ "ISpaceId" ]
		},
		webAppSettings : {
			name : "Map",
			arguments : [ "String", "WebAppSettingsUpdateValue" ]
		},
		active : {
			name : "FieldUpdateValue",
			arguments : [ null ]
		},
	});
	return PersonUpdate;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/person/update/PersonUpdate.java
/*
 * Copyright 2017 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.person.update;

import java.util.HashMap;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.id.IPersonId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.space.id.ISpaceId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.webapp.update.WebAppSettingsUpdateValue;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 */
@JsonObject("as.dto.person.update.PersonUpdate")
public class PersonUpdate implements IUpdate, IObjectUpdate<IPersonId>
{

    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IPersonId userId;

    @JsonProperty
    private FieldUpdateValue<ISpaceId> spaceId = new FieldUpdateValue<ISpaceId>();

    @JsonProperty
    private FieldUpdateValue<Boolean> active = new FieldUpdateValue<Boolean>();

    @JsonProperty
    private Map<String, WebAppSettingsUpdateValue> webAppSettings;

    public IPersonId getUserId()
    {
        return userId;
    }

    public void setUserId(IPersonId personId)
    {
        this.userId = personId;
    }

    @Override
    @JsonIgnore
    public IPersonId getObjectId()
    {
        return getUserId();
    }

    @JsonIgnore
    public void setSpaceId(ISpaceId spaceId)
    {
        this.spaceId.setValue(spaceId);
    }

    @JsonIgnore
    public FieldUpdateValue<ISpaceId> getSpaceId()
    {
        return spaceId;
    }

    @JsonIgnore
    public WebAppSettingsUpdateValue getWebAppSettings(String webAppId)
    {
        if (webAppSettings == null)
        {
            webAppSettings = new HashMap<String, WebAppSettingsUpdateValue>();
        }

        WebAppSettingsUpdateValue updateValue = webAppSettings.get(webAppId);

        if (updateValue == null)
        {
            updateValue = new WebAppSettingsUpdateValue();
            webAppSettings.put(webAppId, updateValue);
        }

        return updateValue;
    }

    public Map<String, WebAppSettingsUpdateValue> getWebAppSettings()
    {
        return webAppSettings;
    }

    @JsonIgnore
    public FieldUpdateValue<Boolean> isActive()
    {
        return active;
    }

    @JsonIgnore
    public void activate()
    {
        active.setValue(true);
    }

    @JsonIgnore
    public void deactivate()
    {
        active.setValue(false);
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("userId", userId).toString();
    }

}
